libraries/Microsoft.Bot.Connector/ConnectorClientEx.cs

```diff
@@ -27,54 +31,50 @@ public ConnectorClient(Uri baseUri, string microsoftAppId = null, string microso
+        /// <summary>
+        /// Create a new instance of the ConnectorClient class
+        /// </summary>
+        /// <param name="baseUri">Base URI for the Connector service</param>
+        /// <param name="credentials">Credentials for the Connector service</param>
+        /// <param name="addJwtTokenRefresher">(DEPRECATED)</param>
+        /// <param name="handlers">Optional. The delegating handlers to add to the http client pipeline.</param>
+        public ConnectorClient(Uri baseUri, MicrosoftAppCredentials credentials, bool addJwtTokenRefresher = true, params DelegatingHandler[] handlers)
+            : this(baseUri, handlers)
         {
-            var httpClient = new HttpClient();
-            // The Schema version is 3.1, put into the Microsoft-BotFramework header
-            // https://github.com/Microsoft/botbuilder-dotnet/issues/471
-            httpClient.DefaultRequestHeaders.UserAgent.Add(new ProductInfoHeaderValue("Microsoft-BotFramework", "3.1"));
-
-            // The client SDK version is coupled to the version number of the package. 
-            httpClient.DefaultRequestHeaders.UserAgent.Add(new ProductInfoHeaderValue($"(BotBuilder .Net/{typeof(ConnectorClient).GetTypeInfo().Assembly.GetName().Version})"));
-            httpClient.DefaultRequestHeaders.ExpectContinue = false;
-            return httpClient;
-        });
+            this.Credentials = credentials;
+        }
 
-        public bool UseSharedHttpClient
+        /// <summary>
+        /// Create a new instances of the ConnectorClient.
+        /// </summary>
+        /// <param name="baseUri">Base URI for the Connector service</param>
+        /// <param name="credentials">Credentials for the Connector service</param>
+        /// <param name="httpClientHandler">The httpClientHandler used by http client</param>
+        /// <param name="addJwtTokenRefresher">(DEPRECATED)</param>
+        /// <param name="handlers">Optional. The delegating handlers to add to the http client pipeline.</param>
+        public ConnectorClient(Uri baseUri, MicrosoftAppCredentials credentials, HttpClientHandler httpClientHandler, bool addJwtTokenRefresher = true, params DelegatingHandler[] handlers)
+            : this(baseUri, httpClientHandler, handlers)
         {
-            get { return this._originalHttpClient != null; }
-            set
-            {
-                if (value == false)
-                {
-                    this.HttpClient = this._originalHttpClient;
-                }
-                else
-                {
-                    if (this._originalHttpClient == null)
-                    {
-                        this._originalHttpClient = this.HttpClient;
-                        this.HttpClient = g_httpClient.Value;
-                    }
-                }
-            }
+            this.Credentials = credentials;
         }
 
         partial void CustomInitialize()
```

Comment:
I have... never used a partial method before. 

We should consider changing this to a more well known pattern if that's practical.

Reply:
this is the approved way of doing this for autorest generated code.

---
In reply to: [198678862](https://github.com/Microsoft/botbuilder-dotnet/pull/702#discussion_r198678862) [](ancestors = 198678862)